CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -68,6 +73,68 @@ internal void UpdateFileHeader(TextDocument textDocument)
             }
         }
 
+        /// <summary>
+        /// Replace the variables in the header if any
+        /// </summary>
+        /// <param name="textDocument"></param>
+        internal string ReplaceVariables(TextDocument textDocument, string settingsFileHeader)
+        {
+            var outputSettingsFileHeader = settingsFileHeader;
+
+            foreach (var variable in _variables)
+            {
+                // If the file header doesn't contain this variable, do nothing...
+                if (!settingsFileHeader.Contains(variable))
+                {
+                    continue;
+                }
+
+                var variableValue = string.Empty;
+                switch (variable)
+                {
+                    case "$USER_LOGIN$":
+                        {
+                            variableValue = Environment.UserName;
+                            break;
+                        }
+                    case "$SOLUTION$":
+                        {
+                            if (_package.IDE.Solution != null)
+                            {
+                                variableValue = Path.GetFileNameWithoutExtension(_package.IDE.Solution.FullName);
+                            }
+
+                            break;
+                        }
+                    case "$PROJECT$":
+                        {
+                            try
+                            {
+                                if (_package.IDE.Solution != null)
+                                {
+                                    variableValue = _package.IDE.Solution.Projects.Item(1).Name;
```

Comment:
I believe this is just always going to give you the name of the first project in your solution.. not the project for the current file.  In other words if you have three projects in your solution A, B and C.. and your file is C.. shouldn't this resolve C?  The best way to approach that would be to walk upwards from the textDocument.ProjectItem

Reply:
Yep you are right, I'll fix that! Thanks for spotting that !